# Sample luxembourgish instruction tuning set

In this notebook, we randomly sample 200 rows from the original luxembourgish instruction tuning set, akin to the sampleing done during the reward process. We will sample 80% from the RTL data and 20% from the wiki data, keeping approximately the distribution in the original dataset, ending up with 160 rows generated from RTL data and 40 rows generated from Wiki data.

In [1]:
# Mount Google Drive
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [2]:
# read in the english csv file
import pandas as pd
df = pd.read_csv('/content/drive/My Drive/Master Thesis/instruction_tuning_set.csv')
df

instruction  \
0     Ginn déi wichtegst Informatiounen iwwer d'Debo...   
1     Wat ass déi bekanntst Performance vun der Debo...   
2     Wat ass den Hannergrond vun der Performance vu...   
3     Wat huet d'Deborah De Robertis den 6. Mee 2024...   
4     Wéi gouf d'Deborah De Robertis den 29. Mee 202...   
...                                                 ...   
2415  Wat war den 1. August 2011 de Kontext vum Staa...   
2416  Wéivill soll d'Scholdegrenz an den USA erhéijt...   
2417              Wat sinn d'Reaktiounen op den Accord?   
2418  Wat sinn d'Konsequenze, wann den Accord net ug...   
2419          Wéi hu d'Boursen op den Accord reagéiert?   

                                               response  \
0     D'Deborah De Robertis ass eng lëtzebuergesch-f...   
1     Déi bekanntst Performance vun der Deborah De R...   
2     Den 31. August 2018 huet d'Deborah De Robertis...   
3     Den 6. Mee 2024 huet d'Deborah De Robertis am ...   
4     Den 29. Mee 2024 gouf d'Deborah De Robertis vi...   
...                                                 ...   
2415  Den 1. August 2011 hat de President Barack Oba...   
2416  D'Scholdegrenz soll an zwou Etappe ëm op manns...   
2417  Net jiddereen ass mat dësem Kompromëss d'Accor...   
2418  Wann den Accord net ugeholl gëtt, kéinten d'US...   
2419  D'Boursen hu weltwäit positiv op den Accord re...   

                                                   text  
0     The following provides an instruction in form ...  
1     The following provides an instruction in form ...  
2     The following provides an instruction in form ...  
3     The following provides an instruction in form ...  
4     The following provides an instruction in form ...  
...                                                 ...  
2415  The following provides an instruction in form ...  
2416  The following provides an instruction in form ...  
2417  The following provides an instruction in form ...  
2418  The following provides an instruction in form ...  
2419  The following provides an instruction in form ...  

[2420 rows x 3 columns]

# Sample Data

We split the dataset into two parts. The first part contains all rows that were generated from the wikipedia data, rows 0 to 489 and the second part, contaning all rows that were generated from the RTL data, rows 490 to 2419. We then randomly sample 40 rows from the wiki data and 160 rows from the RTL data, maintaining the 0.2/0.8 split from the original dataset.

In [3]:
wiki_df = df.iloc[:490]
rtl_df = df.iloc[490:]
print(len(wiki_df))
print(len(rtl_df))

490
1930


In [4]:
# random sample 40 rows from wiki data
sampled_wiki_df = wiki_df.sample(n=40, random_state=42)
sampled_wiki_df = sampled_wiki_df.reset_index(drop=True) # reset index and drop orignal index
sampled_wiki_df

instruction  \
0                  Wat ass de hellste Stär am Dorado?   
1   Wat ass mam Edikt vun Nantes geschitt, nodeem ...   
2   Wann huet de Devolutiounskrich ugefaangen a wé...   
3           Wann an wou ass de Max Ophüls gestuerwen?   
4   Wat ass mam Gebai vum Schumacher geschitt, nod...   
5   Wat sinn déi essenziell Aminosaieren? Nimm déi...   
6   Ginn mir d'Nimm vun de fréiere Gemenge déi 197...   
7   Ginn mir d'Fläch vun der Polvermillen a wat Pr...   
8   Wéini huet de Victor Tedesco seng Prisongstrof...   
9   Wéi vill Filmer huet d'Cinémathèque de la Vill...   
10  Wéi gouf de Leo VIII. zum zweeten Mol zum Poopst?   
11  Wat fir Concoursen organiséiert d'Musekshéichs...   
12  Wat war d'Haaptaarbechtsgebitt vum Martin Schw...   
13  Wéini ass de Mathias Thinnes an d'Pensioun gaa...   
14  Ginn d'Geschicht vun der Kathedral Notre-Dame ...   
15  Ginn mir den Numm vun der lëtzebuergescher Doo...   
16  Wat war den Devolutiounskrich a wéi huet e sec...   
17  Wat waren d'Ziler vum Kinnek Louis XIV. wärend...   
18        Ginn d'Datum vun der Wei vun der Kathedral.   
19  Wat war d'Legitimitéit vun der hellenistescher...   
20  Wéini huet de Jean-Baptiste Kaeuffer seng Par ...   
21  Wéini gouf den Heinrich Rütten Landrat vum Kre...   
22  Wat huet de Martin Schwarzschild zesumme mam L...   
23  Ginn mir den Numm vum Resistenzler deen den 2....   
24  Wat war d'Nominatioun vum Jean-Baptiste Kaeuff...   
25  Wéini huet de Jean-Baptiste Kaeuffer op d'Drän...   
26  Wat gouf aus dem Schrëftstellerheem, nodeems d...   
27   Ginn mir eng kuerz Biographie vum Marcel Gilles.   
28  Wat war d'Zil vum Jayson Sherlock, wéi hien Ho...   
29         Wéi déif ass den Erieséi am Duerchschnëtt?   
30           Wou gouf d'Serie Dawson's Creek gefilmt?   
31  Wat ass den héchste Punkt an der Gemeng Roussy...   
32  Gitt d'Lëscht vun den 20 (+ 1) Aminosaieren dé...   
33  Wat ass d'Formel fir d'Waasserquantitéit ze be...   
34  Wat ass d'Beschreiwung vum Wope vun der Gemeng...   
35  Beschreift den musekalesche Stil vun Plaguewie...   
36  Ginn déi wichtegst Informatiounen iwwer d'Debo...   
37                           Wou läit Nidderkäerjeng?   
38  An wat fir enger Entreprise investéiert Virgin...   
39  Wat war de Sujet vun der Gedenkmënz zum 50. An...   

                                             response  \
0   Den α  Doradus, den hellste Stär am Dorado, as...   
1   Den Edikt vun Nantes gouf leider net "perpétue...   
2   De Devolutiounskrich huet am Fréijouer 1667 ug...   
3   De Max Ophüls ass de 26. Mäerz 1957 zu Hamburg...   
4   De Policemusée vun der Asbl 'Le musée de la po...   
5   Déi essenziell Aminosaieren sinn: Histidin, Is...   
6   D'fréier Gemenge Beho, Bovigny, Cherain, Limer...   
7   D'Polvermillen huet eng Fläch vu 24,82 ha, wat...   
8   No der gescheiterter Expeditioun vun Exilbelsc...   
9   D'Cinémathèque de la Ville de Luxembourg versu...   
10  De Leo VIII. gouf zum zweeten Mol zum Poopst n...   
11  D'Musekshéichschoul Franz Liszt Weimar organis...   
12  Säin Haaptaarbechtsgebitt war d'Stärenentwéckl...   
13                              Den 1. September 1978   
14  D'Kathedral Notre-Dame vu Lëtzebuerg gouf tësc...   
15                                      Plaguewielder   
16  Den Devolutiounskrich war e militäresche Konfl...   
17  D'Ziler vum Kinnek Louis XIV. waren, sech terr...   
18  Den 8. Dezember 1963 huet de Bëschof Léon Lomm...   
19  D'hellenistesch Monarchie war perséinlech. Dat...   
20  De Jean-Baptiste Kaeuffer huet seng Par de 25....   
21                                   De 5. Juli 1944.   
22  Zesumme mam Lyman Spitzer an James Van Allen h...   
23                                       Émile Benoit   
24  De Jean-Baptiste Kaeuffer gouf den 11. August ...   
25  De Jean-Baptiste Kaeuffer huet am Oktober 1797...   
26  An dem Schrëftstellerheem gouf d'Handwierkersc...   
27  De Marcel Victor Gilles, gebuer den 29. Juni 1...   
28  D'Zil vum Jayson Sherlock war et eng chrëschtl.

In [5]:
# now same for RTL data, but with 160 rows
sampled_rtl_df = rtl_df.sample(n=160, random_state=42)
sampled_rtl_df = sampled_rtl_df.reset_index(drop=True)
sampled_rtl_df

instruction  \
0     Wéini huet de Marco Mille bei Siemens geschafft?   
1    Wat ass de Resultat vun der grouss ugeluechte ...   
2    Gitt mir d'Zuel vun de Jonken zu Lëtzebuerg, d...   
3    Wat huet d'Proprietärin vun engem Déier iwwer ...   
4    Wéi vill Agencen sinn zanter 2022 fir d'Format...   
..                                                 ...   
155  Gëtt et och aner Méiglechkeeten fir politesch ...   
156            Wann a wou fiert de City Night Bus CN1?   
157  Wat war den Haaptthema vun de Berodunge vum Co...   
158  Wat ass d'Rôle vun der ierfgroussherzoglecher ...   
159  Wien huet déi Coupe am Basket am Joer 2004 gew...   

                                              response  \
0    Ereéischt am Mäerz 2010 geet de Marco Mille be...   
1    D'Police huet 20 Avertissement taxé an ee Prot...   
2    Am Joer 2019 waren 9,75 Prozent vun de Kanner ...   
3    D'Proprietärin huet gesot, d'Ugeklote hätt net...   
4    Zanter 2022 sinn sechs Agencen fir d'Formation...   
..                                                 ...   
155  Jo, et gëtt och aner Méiglechkeeten fir polite...   
156  De City Night Bus CN1 geet vun der Uewerstad i...   
157  Den Haaptthema war d'Flüchtlingsproblematik, b...   
158  Mir liewen wierklech a ganz komplizéierten Zäi...   
159  Déi Musel Pikes hunn déi Coupe am Basket am Jo...   

                                                  text  
0    The following provides an instruction in form ...  
1    The following provides an instruction in form ...  
2    The following provides an instruction in form ...  
3    The following provides an instruction in form ...  
4    The following provides an instruction in form ...  
..                                                 ...  
155  The following provides an instruction in form ...  
156  The following provides an instruction in form ...  
157  The following provides an instruction in form ...  
158  The following provides an instruction in form ...  
159  The following provides an instruction in form ...  

[160 rows x 3 columns]

In [6]:
# Now we merge the dataframes back together
df = pd.concat([sampled_wiki_df, sampled_rtl_df])
df = df.reset_index(drop=True)
df

instruction  \
0                   Wat ass de hellste Stär am Dorado?   
1    Wat ass mam Edikt vun Nantes geschitt, nodeem ...   
2    Wann huet de Devolutiounskrich ugefaangen a wé...   
3            Wann an wou ass de Max Ophüls gestuerwen?   
4    Wat ass mam Gebai vum Schumacher geschitt, nod...   
..                                                 ...   
195  Gëtt et och aner Méiglechkeeten fir politesch ...   
196            Wann a wou fiert de City Night Bus CN1?   
197  Wat war den Haaptthema vun de Berodunge vum Co...   
198  Wat ass d'Rôle vun der ierfgroussherzoglecher ...   
199  Wien huet déi Coupe am Basket am Joer 2004 gew...   

                                              response  \
0    Den α  Doradus, den hellste Stär am Dorado, as...   
1    Den Edikt vun Nantes gouf leider net "perpétue...   
2    De Devolutiounskrich huet am Fréijouer 1667 ug...   
3    De Max Ophüls ass de 26. Mäerz 1957 zu Hamburg...   
4    De Policemusée vun der Asbl 'Le musée de la po...   
..                                                 ...   
195  Jo, et gëtt och aner Méiglechkeeten fir polite...   
196  De City Night Bus CN1 geet vun der Uewerstad i...   
197  Den Haaptthema war d'Flüchtlingsproblematik, b...   
198  Mir liewen wierklech a ganz komplizéierten Zäi...   
199  Déi Musel Pikes hunn déi Coupe am Basket am Jo...   

                                                  text  
0    The following provides an instruction in form ...  
1    The following provides an instruction in form ...  
2    The following provides an instruction in form ...  
3    The following provides an instruction in form ...  
4    The following provides an instruction in form ...  
..                                                 ...  
195  The following provides an instruction in form ...  
196  The following provides an instruction in form ...  
197  The following provides an instruction in form ...  
198  The following provides an instruction in form ...  
199  The following provides an instruction in form ...  

[200 rows x 3 columns]

In [7]:
# save as csv file
df.to_csv('/content/drive/My Drive/Master Thesis/sampled_instruction_tuning_set.csv', index=False)